cmd/parca-agent/main.go

```diff
@@ -442,34 +454,57 @@ func run(logger log.Logger, reg *prometheus.Registry, flags flags) error {
 		return fmt.Errorf("failed to open procfs: %w", err)
 	}
 
+	// Run group for process tree.
 	psTree := process.NewTree(logger, pfs, flags.Profiling.Duration)
 	{
+		logger := log.With(logger, "group", "process_tree")
 		ctx, cancel := context.WithCancel(ctx)
 		g.Add(func() error {
+			level.Debug(logger).Log("msg", "starting")
+			defer level.Debug(logger).Log("msg", "stopped")
+
 			return psTree.Run(ctx)
 		}, func(error) {
+			level.Debug(logger).Log("msg", "cleaning up")
+			defer level.Debug(logger).Log("msg", "cleanup finished")
 			cancel()
 		})
 	}
 
+	// Run group for metadata discovery.
 	discoveryMetadata := metadata.ServiceDiscovery(logger, discoveryManager.SyncCh(), psTree)
 	{
+		logger := log.With(logger, "group", "metadata_discovery")
 		ctx, cancel := context.WithCancel(ctx)
 		g.Add(func() error {
+			level.Debug(logger).Log("msg", "starting")
+			defer level.Debug(logger).Log("msg", "stopped")
+
 			return discoveryMetadata.Run(ctx)
 		}, func(error) {
+			level.Debug(logger).Log("msg", "cleaning up")
+			defer level.Debug(logger).Log("msg", "cleanup finished")
 			cancel()
 		})
 	}
 
+	curr, max, err := rlimitNOFILE()
+	if err != nil {
+		return fmt.Errorf("failed to get rlimit NOFILE: %w", err)
+	}
+	level.Info(logger).Log("msg", "rlimit", "cur", curr, "max", max)
+
+	ofp := objectfile.NewPool(logger, reg, curr) // Probably we need a little less than this.
```

Comment:
We might be sharing resources with other applications, ideally, this number should always be lower than the maximum. I think there are other mechanisms here to prevent it from reaching the max passed fds, but this should not be as high.

Potentially we can have an initial check on start-up to check that the fd limit is not too low + safety mechanism here to ensure we don't use too many fds?

Either way, adding this as a to-check once this is released to prod/demo